src/runtime/CollectionWrappers/ListWrapper.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+
+namespace Python.Runtime.CollectionWrappers
+{
+    internal class ListWrapper<T> : SequenceWrapper<T>, IList<T>
+    {
+        public ListWrapper(PyObject pyObj) : base(pyObj)
+        {
+
+        }
+
+        public T this[int index]
+        {
+            get
+            {
+                var item = Runtime.PyList_GetItem(pyObject.Handle, index);
+                var pyItem = new PyObject(item);
+
+                if (!Converter.ToManaged(pyItem.Handle, typeof(T), out object obj, true))
+                    Runtime.CheckExceptionOccurred();
+
+                return (T)obj;
+            }
+            set
+            {
+                IntPtr pyItem = Converter.ToPython(value, typeof(T));
+                if (pyItem == IntPtr.Zero)
+                    throw new Exception("failed to set item");
```

Comment:
Perhaps `InvalidCastException`. Also, might need to supply the cause (e.g. underlying `PythonException`).